src/core.ts

```diff
@@ -192,6 +184,14 @@ function upload(payload: string, onSuccess?: UploadCallback, onFailure?: UploadC
     defaultUpload(payload, onSuccess, onFailure);
   }
   sentBytesCount += payload.length;
+  if (state === State.Activated && sentBytesCount > config.totalLimit) {
```

Comment:
Because upload is invoked from teardown() method to upload remaining events. Is we don't check the state, then this would call circular call sequence.

---
In reply to: [136483683](https://github.com/Microsoft/clarity-js/pull/28#discussion_r136483683) [](ancestors = 136483683)